x/data/client/tx.go

```diff
@@ -25,22 +31,119 @@ func TxCmd() *cobra.Command {
 	return cmd
 }
 
+// MsgAnchorDataCmd creates a CLI command for Msg/AnchorData.
 func MsgAnchorDataCmd() *cobra.Command {
 	cmd := &cobra.Command{
-		Use: "send [from_key_or_address] [to_address] [amount]",
-		Short: `Send funds from one account to another. Note, the'--from' flag is
-ignored as it is implied from [from_key_or_address].`,
-		Args: cobra.ExactArgs(3),
+		Use: "anchor [cid]",
+		Short: `Anchors a piece of data to the blockchain based on its secure
+		hash, effectively providing a tamper resistant timestamp.`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			cid, err := gocid.Decode(args[0])
+			if err != nil {
+				return err
+			}
+
+			msg := data.MsgAnchorDataRequest{
+				Sender: clientCtx.GetFromAddress().String(),
+				Cid:    cid.Bytes(),
+			}
+			svcMsgClientConn := &util.ServiceMsgClientConn{}
+			msgClient := data.NewMsgClient(svcMsgClientConn)
+			_, err = msgClient.AnchorData(context.Background(), &msg)
+			if err != nil {
+				return err
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), svcMsgClientConn.Msgs...)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// MsgSignDataCmd creates a CLI command for Msg/SignData.
+func MsgSignDataCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sign [cid]",
+		Short: `Sign an arbitrary piece of data on the blockchain.`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			clientCtx, err := client.ReadTxCommandFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			cid, err := gocid.Decode(args[0])
+			if err != nil {
+				return err
+			}
+
+			msg := data.MsgSignDataRequest{
+				Signers: []string{clientCtx.GetFromAddress().String()},
```

Comment:
I'm still not sure what's the best UX for CLI users. Say 3 signers want to sign the same CID:
- either they run this CLI command each individually
- or this CLI command just generates a tx file, and people pass around that tx file and each signs the file with their CLI (via x/auth's Sign command)

I prefer the first UX, so put it here for now.